### ISSUE TITLE
GraphQl: cant disable operations with yaml

### ISSUE THREAD

**Issue by OndrosI** [api-platform/api-platform#2341]
Hi iam using api/platform 3.0.7 and i want to modify graphql operations with yaml.

My yaml looks like this:

REST: i want only GET method
GRAPHQL: i want enable only Query and Create mutation

```
Namespace\To\Entity:
    operations:
        ApiPlatform\Metadata\GetCollection: ~
        ApiPlatform\Metadata\Get: ~
    graphQlOperations:
        ApiPlatform\Metadata\GraphQl\Query: ~
        ApiPlatform\Metadata\GraphQl\Mutation:
            name: create
```

with REST ("operation" namespace) it works fine and i can see only GET methods. But with graphql i still can do collenctionQuery and all other mutations. 

What am i doing wrong?

Thx for answer

**Comment by alanpoulain** [comment]
Its need to be written like this:
```yaml
Namespace\To\Entity:
    operations:
        ApiPlatform\Metadata\GetCollection: ~
        ApiPlatform\Metadata\Get: ~
    graphQlOperations:
        queries:
            class: ApiPlatform\Metadata\GraphQl\Query
        mutations:
            class: ApiPlatform\Metadata\GraphQl\Mutation
            name: create
````

@vincentchalamon could you explain why it is like this?

**Comment by vincentchalamon** [comment]
It has been build to separate queries, mutations and subscriptions because of the common class for each of these operations.

According to the [YamlExtractor](https://github.com/api-platform/core/blob/main/src/Metadata/Extractor/YamlResourceExtractor.php#L288-L334), you should be able to declare it as following:
```yaml
Namespace\To\Entity:
    operations:
        ApiPlatform\Metadata\GetCollection: ~
        ApiPlatform\Metadata\Get: ~
    graphQlOperations:
        queries:
            - class: ApiPlatform\Metadata\GraphQl\Query
              name: foo
            - class: ApiPlatform\Metadata\GraphQl\Query
              name: bar
        mutations:
            - class: ApiPlatform\Metadata\GraphQl\Mutation
              name: create
```

> Note that each query use the same class, which mean the class fqcn cannot be the index.

**Comment by alanpoulain** [comment]
@vincentchalamon I think it's cumbersome compared to REST, don't you think?

Why not just:

```yaml
Namespace\To\Entity:
    graphQlOperations:
        - class: ApiPlatform\Metadata\GraphQl\Query
          name: foo
        - class: ApiPlatform\Metadata\GraphQl\Query
          name: bar
        - class: ApiPlatform\Metadata\GraphQl\Mutation
          name: create
```

**Comment by vincentchalamon** [comment]
Indeed, it can be simplified! Any PR is welcome, but without BC break, and with XML support as well.

**Comment by vincentchalamon** [comment]
Also, maybe the following configuration is possible:
```yaml
Namespace\To\Entity:
    graphQlOperations:
        queries:
            # "class" is optional and default is Query::class
            - name: foo
            - name: bar
        mutations:
            # "class" is optional and default is Mutation::class
            - name: baz
        subscriptions:
            # "class" is optional and default is Subscription::class
            - name: bar
```

**Comment by alanpoulain** [comment]
IMO we can have a BC break here, since it is just released and not documented.

**Comment by OndrosI** [comment]
> operations:
>         ApiPlatform\Metadata\GetCollection: ~
>         ApiPlatform\Metadata\Get: ~

Thank you, this works fine. 

It would be nice to update documentation in next hotfix.

**Comment by klobastov** [comment]
```
Namespace\To\Entity:
  graphQlOperations:
    queries:
      - class: Namespace\To\Entity
      - class: Namespace\To\Entity
        collection: true
```
api-platform/core 3.1.6

Operation class \"queries\" does not exist

**Comment by vincentchalamon** [comment]
@OndrosI did you try with the latest version of API Platform? This has also been fixed in 3.0.11 (cf. https://github.com/api-platform/core/pull/5333):
```yaml
Namespace\To\Entity:
    operations:
        ApiPlatform\Metadata\GetCollection: ~
        ApiPlatform\Metadata\Get: ~
    graphQlOperations:
        ApiPlatform\Metadata\GraphQl\Query: ~
        ApiPlatform\Metadata\GraphQl\Mutation:
            name: create
```

**Comment by OndrosI** [comment]
@vincentchalamon yea it works, thx. If u can pls update documentation.
